libs/execution_context/ebpf_program.c

```diff
@@ -1478,15 +1477,26 @@ _ebpf_helper_id_to_index_compare(const void* lhs, const void* rhs)
 
 /**
  * @brief Compute the hash of the program info and compare it with the hash stored in the program. If the hash does not
- * match then the program was verified against the wrong program info.
+ * match then the program was verified against the wrong program info. If the hash is not present then store the hash
+ * in the program so it can be compared when the program information provider reattaches.
+ *
+ * Notes on why this works:
+ * 1) The user application creates a ebpf_program_t object and set the program type.
+ * 2) During initialization, the program binds to the program information provider.
+ * 3) During the attach callback, the program information is hashed and stored.
```

Comment:
One can load a program without attaching, it and verification has to happen on load.  So why is 3 before 4?

Reply:
Need to clarify what attach means in this context. This is the NMR attach to the program information provider (not the BPF attach point).